contracts/base/LenderInfo.sol

```diff
@@ -0,0 +1,216 @@
+        
+        if (accruedInterest >= amountToWithdraw) {
+            lenderAccounts[recipient].totalAccruedInterest = accruedInterest.sub(amountToWithdraw);
+        } else {
+            amountToWithdraw = lenderAccounts[recipient].totalAccruedInterest;
+            lenderAccounts[recipient].totalAccruedInterest = 0;
+        }
+        lenderAccounts[recipient].lastBlockAccrued = this.getCurrentBlockNumber();
+
+        emit AccruedInterestWithdrawn(recipient, amountToWithdraw);
+
+        return amountToWithdraw;
+    }
+
+    function getCurrentBlockNumber()
+        external
+        view
+        returns (uint256 blockNumber)
+    {
+        return block.number;
+    }
+
+    /** Internal Functions */
+
+    /**
+        Updates the accrued interest for a specific lender.
+    
+        @param lender address.
+        @return the updated accrued intereset for the lender address.
+     */
+    function updateAccruedInterestFor(address lender)
+        internal
+        returns (uint256)
+    {
+        // Get last block accrued
+        uint256 previousBlockAccruedInterest = lenderAccounts[lender].lastBlockAccrued;
+        uint256 currentBlockNumber = this.getCurrentBlockNumber();
+
+        // Update last block accrued.
+        lenderAccounts[lender].lastBlockAccrued = currentBlockNumber;
+
+        // Update total accrued interest.
+        lenderAccounts[lender].totalAccruedInterest = calculateNewAccruedInterestFor(
+                                                            lenderAccounts[lender].totalAccruedInterest,
+                                                            previousBlockAccruedInterest,
+                                                            currentBlockNumber,
+                                                            getZDaiBalanceOf(lender)
+                                                        );
+
+        emit AccruedInterestUpdated(
+            lender,
+            lenderAccounts[lender].lastBlockAccrued,
+            lenderAccounts[lender].totalAccruedInterest
+        );
+
+        return lenderAccounts[lender].totalAccruedInterest;
+    }
+
+    function calculateNewAccruedInterestFor(
+        uint256 currentAccruedInterest,
+        uint256 previousBlockAccruedInterest,
+        uint256 currentBlockNumber,
+        uint256 currentZDaiBalance
+    )
+        internal
+        pure
+        returns (uint256 newAccruedInterest)
+    {
+        uint256 blocksDifference = currentBlockNumber.sub(previousBlockAccruedInterest);
+        return currentAccruedInterest.add(blocksDifference.mul(currentZDaiBalance));
+    }
+
+    function areAddressesEqual(address leftAddress, address rightAddress)
+        internal
+        view
+        returns (bool)
+    {
+        return leftAddress.isEqualTo(rightAddress);
+    }
+
+    /** Private Functions */
+
+    function getZDaiBalanceOf(address anAddress) private view returns (uint256 balance) {
```

Comment:
Again this function to me seems like it doesnt add anything - as it's not doing any form of calculation etc. It's doing 1 simple call . What is your reason for putting it inside its own function?

Reply:
As I mentioned before, mainly to make the code simpler/readable/testable.